### ISSUE TITLE
Mouseover and click event clash when element changes

### ISSUE THREAD

**Issue by BCotrim** [angular/angular#8996]
* **I'm submitting a ...**
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support request here, see note at the top of this template.

**Current behavior**
When we have a mouseover and a click event on the same element, if the mouseover forces the element to be changed (e.g. a change on it's text value), the click event stops being fired, and a mouseover is fired instead.
Note: The test is flaky, sometimes it works as intended.

**Expected/desired behavior**
The "click" and "mouseover" events should be triggered correctly.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** via
https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5).
https://plnkr.co/edit/oUV5cwaBBhjjk6xqI1Ql?p=preview

* **What is the expected behavior?**
The "click" and "mouseover" events should be triggered correctly.

* **What is the motivation / use case for changing the behavior?**
It is a bug.

* **Please tell us about your environment:**
Tested on plunker.

* **Angular version:** 2.0.0-beta.X 
 2.0.0-beta.17

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ] 
Chrome Version 50.0.2661.102 (64-bit)
Safari Version 9.1.1 (11601.6.17)
Edge
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5 | Dart] 
Typescript

**Comment by pkozlowski-opensource** [comment]
I'm not sure how real-life this use-case is (did you discover this in your app or just pushing the limits of the framework?) but if you want to make it work you need to use a track by function for ngFor: https://plnkr.co/edit/0TlAVKO9qKN4pwisZkCU?p=preview

If you don't Angular can't match numbers in your array with the corresponding DOM elements so when you mouseover a given element it is destroyed before it has a chance to handle click event.

**Comment by BCotrim** [comment]
Thanks for your reply.
I got this from my app. It's not really easy to explain the use case because it's a lot more complicated than the plunker I did. 
I have parent Component and 2 child components. 
The child components (that represent a date grid) have events for click and mouseover on the grid. 
The parent component handles the triggered events from both child components and changes a value on both that cause them to recalculate some other values inside the component.

Regarding the solution it absolutely makes sense, thank you for your reply it really helped me.
Sorry I opened this without being a bug.
